core/src/main/scala/memeid/UUID.scala

```diff
@@ -71,59 +77,113 @@ sealed trait UUID extends Comparable[UUID] {
    * The variant number has the following meaning:
    *
    * - '''0''': Reserved for NCS backward compatibility
+   *
    * - '''2''': [[https://tools.ietf.org/html/rfc4122 IETF RFC 4122]]
+   *
    * - '''6''': Reserved, Microsoft Corporation backward compatibility
+   *
    * - '''7''': Reserved for future definition
    *
    * Interoperability, in any form, with variants other than the one
    * defined here is not guaranteed, and is not likely to be an issue in
    * practice.
    *
    * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.1]]
+   * @return The variant of this [[UUID]]
    */
   @inline def variant: Int = juuid.variant
 
+  /**
+   * The version number associated with this [[UUID]].  The version
+   * number describes how this [[UUID]] was generated.
+   *
+   * The version number has the following meaning:
+   *
+   * - '''1''':    Time-based UUID
+   *
+   * - '''2''':     DCE security UUID
+   *
+   * - '''3''':    Name-based UUID
+   *
+   * - '''4''':   Randomly generated UUID
+   *
+   * - '''5''': The name-based version that uses SHA-1 hashing
+   *
+   * @see https://tools.ietf.org/html/rfc4122#section-4.1.3
+   * @return The version number of this [[UUID]]
+   */
   @inline def version: Int = juuid.version
 
+  /**
+   * Checks if an [[UUID]] passed as parameter is equal to this [[UUID]]
+   * @param obj object to be checked if it's equal to this [[UUID]]
+   * @return true if both are equals; false otherwise
+   */
   @SuppressWarnings(Array("scalafix:Disable.equals", "scalafix:Disable.Any"))
```

Comment:
I wouldn't add Javadocs for `equals`, `compareTo`, `hashCode` and `toString`, since they are inherited methods from `Object` and `Comparable`.